src/client/api/event.c

```diff
@@ -1028,6 +1028,11 @@ daos_event_init(struct daos_event *ev, daos_handle_t eqh,
 		evx->evx_sched = &eqx->eqx_sched;
 		daos_eq_putref(eqx);
 	} else {
+		if (daos_sched_g.ds_udata == NULL) {
+			D_ERROR("The DAOS client library is not initialized: "
+				DF_RC"\n", DP_RC(-DER_EQ_UNINIT));
+			return -DER_EQ_UNINIT;
+		}
 		evx->evx_ctx = daos_eq_ctx;
```

Comment:
For this code path, before calling tse_task_schedule, dc_task_create() is called which calls this daos_event_init() to check the daos_sched_g.

Since the seg fault is happening in tse_task_schedule, all of this call must after dc_task_create() succeed? is it possible there are cases that tse_task_schedule() is called without first calling dc_task_create() and daos_init()?

Reply:
> For this code path, before calling tse_task_schedule, dc_task_create() is called which calls this daos_event_init() to check the daos_sched_g.
> 
> Since the seg fault is happening in tse_task_schedule, all of this call must after dc_task_create() succeed? is it possible there are cases that tse_task_schedule() is called without first calling dc_task_create() and daos_init()?

Yes, actually the cases where this Segmentation Fault can be found is when calling the daos_obj_open() and all of the following functions without first calling daos_init():
- daos_obj_close()
- daos_obj_punch()
- daos_obj_punch_akeys()
- daos_obj_query_key()
- daos_obj_fetch()
- daos_obj_list_dkey()
- daos_obj_list_akey()
- daos_obj_list_recx()
- daos_obj_verify()

All of the mentioned functions call dc_task_create() resulting in a Seg Fault but with this patch all of them error out with the "DER_EQ_UNINIT" instead.